src/components/Book.js

```diff
@@ -0,0 +1,72 @@
+import { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { CircularProgressbar, buildStyles } from 'react-circular-progressbar';
+import 'react-circular-progressbar/dist/styles.css';
+import PropTypes from 'prop-types';
+
+import Button from './Button';
+import { removeBook } from '../redux/books/books';
+
+const Book = ({
+  id, category, title, author, chapter,
+}) => {
+  const [percent, setPercent] = useState(10);
+
+  const dispatch = useDispatch();
+
+  const handleRemove = () => {
+    dispatch(removeBook({ id }));
+  };
+
+  const handleClick = () => {
+    setPercent((prevState) => (prevState >= 100 ? prevState : prevState + 10));
+  };
+
+  return (
+    <div style={{
+      display: 'flex', alignItems: 'center', justifyContent: 'space-between', background: '#fff',
+    }}
+    >
+      <div>
+        <p>{category}</p>
+        <h2>{title}</h2>
+        <span>{author}</span>
+        <div>
+          <button type="button">Comments</button>
+          <button type="button" onClick={handleRemove}>Remove</button>
+          <button type="button">Edit</button>
+        </div>
+      </div>
+      <div className="progress">
+        <div style={{ width: '4.25rem', height: '4.25rem' }}>
+          <CircularProgressbar value={percent} styles={buildStyles({ pathColor: '#0290ff' })} />
+        </div>
+        <div>
+          <p>
+            {percent}
+            %
+          </p>
+          <p>Completed</p>
+        </div>
+      </div>
+      <div className="chapter">
+        <p>CURRENT CHAPTER</p>
+        <span>
+          Chapter
+          {chapter}
+        </span>
+        <Button text="UPDATE PROGRESS" handleClick={handleClick} />
+      </div>
+    </div>
+  );
+};
+
+Book.propTypes = {
+  id: PropTypes.string.isRequired,
```

Comment:
- [ ] Book details are not been added/ stored correctly. This is likely due to proptype associated with book object. Kindly ensure to fix this issue

**Error:**
![image](https://user-images.githubusercontent.com/12745474/151204127-622841f4-890a-4e9f-a3cd-3b16aa90a74b.png)

Reply:
Solved

![db4](https://user-images.githubusercontent.com/55635977/151211863-3cbd55c5-3067-43ee-98d7-fb7bb21d9a5a.png)